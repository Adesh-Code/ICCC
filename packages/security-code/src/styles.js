const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
