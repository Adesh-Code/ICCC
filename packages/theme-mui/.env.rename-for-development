# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE