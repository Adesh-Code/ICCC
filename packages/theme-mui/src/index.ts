// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
