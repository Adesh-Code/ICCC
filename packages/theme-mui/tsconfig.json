{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
