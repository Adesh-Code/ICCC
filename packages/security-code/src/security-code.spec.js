import React from "react";
import SecurityCode from "./security-code";
import { render } from "@testing-library/react";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const arrange = ({ ...otherProps } = {}) =>
  render(
    <ThemeProvider theme={light}>
      <SecurityCode
        value="123445"
        numInputs={6}
        onChange={jest.fn()}
        {...otherProps}
      />
    </ThemeProvider>
  );

describe("Security code component", () => {
  it("renders", () => {
    const { container } = arrange();
    expect(container).toMatchSnapshot();
  });

  it("renders in a disabled state", () => {
    const { container } = arrange({ disabled: true });
    expect(container.firstChild.firstChild).toHaveClass("Mui-disabled");
  });

  it("renders in an error state", () => {
    const { container } = arrange({ error: true });
    expect(container.firstChild.firstChild).toHaveClass(
      "DhigSecurityCode--error"
    );
  });

  it("renders a dense variant", () => {
    const { container } = arrange({ dense: true });
    expect(container.firstChild.firstChild).toHaveClass(
      "DhigSecurityCode--dense"
    );
  });
});
