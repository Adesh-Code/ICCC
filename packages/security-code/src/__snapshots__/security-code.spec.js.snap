// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security code component renders 1`] = `
<div>
  <div
    style="display: flex; align-items: center;"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd DhigSecurityCode emotion-0"
    >
      <input
        aria-label="Please enter OTP character 1"
        autocomplete="off"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
        maxlength="1"
        style="width: 1em; text-align: center;"
        type="text"
        value="1"
      />
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light MuiDivider-vertical emotion-2"
      />
    </div>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd DhigSecurityCode emotion-0"
    >
      <input
        aria-label="Please enter OTP character 2"
        autocomplete="off"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
        maxlength="1"
        style="width: 1em; text-align: center;"
        type="text"
        value="2"
      />
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light MuiDivider-vertical emotion-2"
      />
    </div>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd DhigSecurityCode emotion-0"
    >
      <input
        aria-label="Please enter OTP character 3"
        autocomplete="off"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
        maxlength="1"
        style="width: 1em; text-align: center;"
        type="text"
        value="3"
      />
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light MuiDivider-vertical emotion-2"
      />
    </div>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd DhigSecurityCode emotion-0"
    >
      <input
        aria-label="Please enter OTP character 4"
        autocomplete="off"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
        maxlength="1"
        style="width: 1em; text-align: center;"
        type="text"
        value="4"
      />
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light MuiDivider-vertical emotion-2"
      />
    </div>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd DhigSecurityCode emotion-0"
    >
      <input
        aria-label="Please enter OTP character 5"
        autocomplete="off"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
        maxlength="1"
        style="width: 1em; text-align: center;"
        type="text"
        value="4"
      />
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light MuiDivider-vertical emotion-2"
      />
    </div>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd DhigSecurityCode emotion-0"
    >
      <input
        aria-label="Please enter OTP character 6"
        autocomplete="off"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd emotion-1"
        maxlength="1"
        style="width: 1em; text-align: center;"
        type="text"
        value="5"
      />
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light MuiDivider-vertical emotion-2"
      />
    </div>
  </div>
</div>
`;
